src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -764,48 +783,75 @@ void SettingsLoader::_parseFragment(const winrt::hstring& source, const std::str
                 // We need to make sure to only call Guid() if HasGuid() is true,
                 // as Guid() will dynamically generate a return value otherwise.
-                const auto guid = profile->HasGuid() ? profile->Guid() : profile->Updates();
-                if (guid != winrt::guid{})
+                auto profile = _parseProfile(OriginTag::Fragment, source, profileJson);
+                if (const auto guid = profile->Guid(); profile->HasGuid() && guid != winrt::guid{})
                 {
-                    _appendProfile(std::move(profile), guid, settings);
+                    if (applyToSettings)
+                    {
+                        _appendProfile(std::move(profile), guid, settings);
+                    }
+                    fragmentSettings->NewProfiles().Append(winrt::make<FragmentProfileEntry>(guid, hstring{ til::u8u16(Json::writeString(styledWriter, profileJson)) }));
+                }
+                else if (const auto guid = profile->Updates(); guid != winrt::guid{})
+                {
+                    if (applyToSettings)
+                    {
+                        _appendProfile(std::move(profile), guid, settings);
+                    }
+                    fragmentSettings->ModifiedProfiles().Append(winrt::make<FragmentProfileEntry>(guid, hstring{ til::u8u16(Json::writeString(styledWriter, profileJson)) }));
                 }
             }
             CATCH_LOG()
         }
     }
 
-    for (const auto& fragmentProfile : settings.profiles)
+    if (applyToSettings)
     {
-        if (const auto updates = fragmentProfile->Updates(); updates != winrt::guid{})
+        for (const auto& fragmentProfile : settings.profiles)
         {
-            if (const auto it = userSettings.profilesByGuid.find(updates); it != userSettings.profilesByGuid.end())
+            if (const auto updates = fragmentProfile->Updates(); updates != winrt::guid{})
             {
-                it->second->AddMostImportantParent(fragmentProfile);
+                if (const auto it = userSettings.profilesByGuid.find(updates); it != userSettings.profilesByGuid.end())
+                {
+                    it->second->AddMostImportantParent(fragmentProfile);
+                }
+            }
+            else
+            {
+                _addUserProfileParent(fragmentProfile);
             }
         }
-        else
+
+        // STAGED COLORS are processed here: we merge them into the partially-loaded
+        // settings directly so that we can resolve conflicts between user-generated
+        // color schemes and fragment-originated ones.
+        for (const auto& [_, fragmentColorScheme] : settings.colorSchemes)
         {
-            _addUserProfileParent(fragmentProfile);
+            if (!_addOrMergeUserColorScheme(fragmentColorScheme))
+            {
+                // Color scheme wasn't added because it conflicted with a non-user created scheme.

```

Comment:
this only captures the case where it was conflicted with another FRAGMENT color scheme. It doesn't know if it failed because it conflicted a built-in color scheme. Or, do those not conflict?

wait, i am confused. :P

Reply:
Actually yeah, I'm going to undo the whole `Conflict` thing. We're not using it and that's an edge case to handle separately and better tbh.